test_data/asm/book/modules.asm

```diff
@@ -22,16 +22,16 @@ mod utils {
 
 machine Main with degree: 8 {
     // use a machine from another module by relative path
-    my_module::Other a;
+    my_module::Other a(8, 8);
```

Comment:
This started as a one-day-before-the-realease-hack and now it's mandatory?

Reply:
In my understanding, the hacky part is the syntax around *how* degrees are passed to sub-machines. I don't think *that* we want to pass degree ranges is controversial?

Reply:
For me it is, shouldn't it somehow at least have a default? What do I care about the degree of a sub-machine, unless I already know that I will call it every single row like the memory machine?

Reply:
I think it's hard to pick a good default heuristically. It depends on the block size and how often we expect to call the machine, and both the user would know better than us.

Reply:
Basically,
- The status quo (pre #2092) is that the default is `min_degree: 2**5, max_degree: 2**MAX_DEGREE_LOG`, where `MAX_DEGREE_LOG` is an environment variable. But using env variables is weird.
- With #2092, we fail if the degree range is unknown after the linker. This is already kind of making it mandatory to set the range, *except* that we have a strange rule in the linker: If the degree of the main machine is static, use that degree for all sub-machines (even if it's already been set differently). That rule is also weird and should be removed IMO. It also doesn't have any effect if the main machine is dynamically sized, which should be the case in most real-world applications.

Reply:
Why not generate the fixed columns only once they are needed? Because of the verification key?

Reply:
Yes, exactly, we only know what's needed at runtime, but at that time, we already need to have committed to the fixed columns in all possible sizes.